src/GitRepository.ts

```diff
@@ -1,46 +1,83 @@
-export type ChangeSummary = {
+import { type Job } from "../mod.ts";
+
+/**
+ * added, removed and modified contiain the respective changed files as a list of paths (strings)
+ */
+export interface ChangeSummary {
   added: string[];
   removed: string[];
   modified: string[];
+  /**
+   * commit hash of commit up until which changes were considered
+   */
   till: string;
-};
+}
+
+export class GitRepository {
+  // non-private members are declared this way and not via the constructor to enable documentation.
+  /**
+   * The uri of the repository that will be cloned into ${workDir}/repository
+   *
+   * Note that authentification via token is only possible for https-uris.
+   *
+   * e.g `"https://github.com/factsmission/ghact.git"`
+   */
+  readonly uri: string;
+  /** Branch to checkout
+   *
+   * e.g `"main"`
+   */
+  readonly branch: string;
+  /**
+   * Where to clone the repository into
+   *
+   * e.g. `"/workdir/repository"`
+   */
+  readonly directory: string;
+
+  private readonly authUri: string;
 
-export default class GitRepository {
-  /** @param repoUrl url of the repository */
   constructor(
-    protected repoUrl: string,
-    protected branch: string,
-    protected token: string | undefined,
-    public workDir: string,
+    uri: string,
+    branch: string,
```

Comment:
why remove the modifier and then having to assign it to a field?

Reply:
Okay, I think that's for them to be readonly, makes sense.

Reply:
`// non-private members are declared this way and not via the constructor to enable documentation.`